src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonDictionaryDefaultConverter.cs

```diff
@@ -0,0 +1,311 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    /// <summary>
+    /// Default base class implementation of <cref>JsonDictionaryConverter{TCollection}</cref> .
+    /// </summary>
+    internal abstract class JsonDictionaryDefaultConverter<TCollection, TValue>
+        : JsonDictionaryConverter<TCollection>
+    {
+        /// <summary>
+        /// When overridden, adds the value to the collection.
+        /// </summary>
+        protected abstract void Add(TValue value, JsonSerializerOptions options, ref ReadStack state);
+
+        /// <summary>
+        /// When overridden, converts the temporary collection held in state.ReturnValue to the final collection.
+        /// This is used with immutable collections.
+        /// </summary>
+        protected virtual void ConvertCollection(ref ReadStack state, JsonSerializerOptions options) { }
+
+        /// <summary>
+        /// When overridden, create the collection. It may be a temporary collection or the final collection.
+        /// </summary>
+        protected virtual void CreateCollection(ref ReadStack state) { }
```

Comment:
We have `Add`, `CreateCollection` and `ConvertCollection` methods also on `JsonIEnumerableConverter`. Would it be good to have both classes to derive form a common class? something like `JsonCollectionConverter`? Although I don't know if that is a good idea.

Reply:
Yes but adding another level doesn't help at run-time since that new base class wouldn't be used. The current code keeps the base classes minimal which could change in the future as you suggest if there happens to be a polymorphic reason to do otherwise.